### PR TITLE
adding managed properties

### PR DESCRIPTION
**Pull request by dmartinol**
Closes/Fixes/Resolves [SRVLOGIC-195](https://issues.redhat.com/browse/SRVLOGIC-195)

**Description:**
Added specification of the `managed properties` concept and of the associated `ConfigMap`

### CODE REVIEW

**Comment by dmartinol** [comment]
@ricardozanini please TAL
Note: I probably need more details about the other `managed properties` from those who developed the discovery and data-index/Job-service features.

**Comment by ricardozanini** [comment]
@nmirasch @wmedvede can you please validate the JS/DI sections?

**Comment by ricardozanini** [comment]
@nmirasch @wmedvede it's been a month, can you guys please take a look?

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc [hunk 1]
```diff
@@ -8,7 +8,13 @@ This document describes how to configure a Workflow service with the {operator_n
 
 == Editing the Workflow Configuration
 
-When the operator deploys the Workflow service, it also creates a `ConfigMap` named after the `SonataFlow` object with the suffix `-props`. For example, if your Workflow name is `greeting`, then the `ConfigMap` name is `greeting-props`.
+When the operator deploys the Workflow service, it also creates two `ConfigMap`s to store the runtime properties:
+
+* The user properties are defined in a `ConfigMap` named after the `SonataFlow` object with the suffix `-props`. For example, if your Workflow name is `greeting`, then the `ConfigMap` name is `greeting-props`.
+* The xref:#managed-props[Managed Properties] are defined in a `ConfigMap` named after the `SonataFlow` object with the suffix `-managed-props`. For example, if your Workflow name is `greeting`, then the `ConfigMap` name is `greeting-props`.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:14]:
```suggestion
* The xref:#managed-props[Managed Properties] are defined in a `ConfigMap` named after the `SonataFlow` object with the suffix `-managed-props`. For example, if your Workflow name is `greeting`, then the `ConfigMap` name is `greeting-props-managed`.

? I don't remember the name exactly but those names differ, right?
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc [hunk 2]
```diff
@@ -8,7 +8,13 @@ This document describes how to configure a Workflow service with the {operator_n
...
+
+Note that managed properties always override any user property using the same key name and cannot be edited by the user: any change would be 
+overwritten by the operator at the next reconciliation cycle.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:17]:
```suggestion
[IMPORTANT]
====
Managed properties always override any user property using the same key name and cannot be edited by the user. Any change would be overwritten by the operator at the next reconciliation cycle.
====
```

**Review by ricardozanini — COMMENTED** [review]
A couple of comments if you don't mind. I'm sorry about the late review.

**Comment by ricardozanini** [comment]
@kaldesai can you take a quick look? Thanks!

**Review by kaldesai — APPROVED** [review]
Changes look good to me

**Review by wmedvede — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc (hunk 1 above)
Comment (domhanak) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:14]: `service-props` and `service-managed-props` is what I see on my deployed example

**Review by nmirasch — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc (hunk 1 above)
Comment (wmedvede) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:14]: The generated name is  service-managed-props

**Comment by ricardozanini** [comment]
@dmartinol can you please rebase so we can check for errors? We've updated our CI to reflect that.

**Review by domhanak — CHANGES_REQUESTED** [review]
Let's resolve the comments and it is good to go.

**Comment by dmartinol** [comment]
Hi Team, it took a very long time to receive your comments: I will try to apply the required changes but please try to avoid such delays. 
In case my updates will not satisfy your comments, please consider pulling my branch and create a new PR.
FYI, @pkliczewski

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc (hunk 1 above)
Comment (dmartinol) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:14]: Updating it, thanks

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc (hunk 2 above)
Comment (dmartinol) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc]: done thanks

**Comment by ricardozanini** [comment]
@dmartinol I'm sorry again for the late review. There's no excuse, that slipped my review list. We will improve our review process.

**Review by domhanak — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc [hunk 3]
```diff
@@ -64,10 +73,18 @@ A few properties cannot be changed in this configuration. Usually, they are alre
 
 |===
 
+Other managed properties include:
+
+* The xref:cloud/quarkus/kubernetes-service-discovery.adoc[Kubernetes service discovery] properties
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:78]:
```suggestion
* The xref:use-cases/advanced-developer-use-cases/service-discovery/kubernetes-service-discovery.adoc[Kubernetes service discovery] properties
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc [hunk 4]
```diff
@@ -64,10 +73,18 @@ A few properties cannot be changed in this configuration. Usually, they are alre
...
+* The xref:data-index/data-index-quarkus-extension.adoc[Data Index location properties]
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:79]:
```suggestion
* The xref:use-cases/advanced-developer-use-cases/data-index/data-index-quarkus-extension.adoc[Data Index location properties]
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc [hunk 5]
```diff
@@ -64,10 +73,18 @@ A few properties cannot be changed in this configuration. Usually, they are alre
...
+* The xref:job-services/quarkus-extensions.adoc[Job Service location properties]
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflows.adoc:80]:
```suggestion
* The xref:use-cases/advanced-developer-use-cases/job-service/quarkus-extensions.adoc[Job Service location properties]
```

**Review by ricardozanini — COMMENTED** [review]
Fixing CI errors.

**Pull request merged**